Add databricks scim command to expose the SCIM api:

areese opened the pull request:
https://docs.databricks.com/dev-tools/api/latest/scim/index.html

Commit:
Add databricks scim command to expose the SCIM api:
https://docs.databricks.com/dev-tools/api/latest/scim/index.html

Commit:
Add scim group

areese committed:
Merge branch 'master' into add-scim-api

areese committed:
Merge branch 'master' into add-scim-api

areese committed:
New files should have the correct copyright headers

areese committed:
Merge branch 'master' into add-scim-api

nfx commented on databricks_cli/utils.py:
```diff
@@ -86,6 +88,18 @@ def decorator(*args, **kwargs):
+    return context_object.debug_mode
+
+
+def debug(name, content):
```
this might not be needed. should be replaced with logger's DEBUG level

nfx commented on databricks_cli/scim/api.py:
```diff
@@ -0,0 +1,269 @@
+
+from databricks_cli.scim.exceptions import ScimError
+from databricks_cli.sdk import ScimService
+from databricks_cli.utils import is_int, debug
```
replace `debug` with log.debug, which is controlled by logging framework

nfx commented on databricks_cli/scim/cli.py:
```diff
@@ -0,0 +1,328 @@
+@debug_option
+@profile_option
+@eat_exceptions
+def scim_group():
```
suggestion: remove SCIM prefix from command groups.

areese replied:
the prefix + _group is the convention across all of the commands:

```
databricks_cli/cluster_policies/cli.py:def cluster_policies_group():  # pragma: no cover
databricks_cli/clusters/cli.py:def clusters_group():  # pragma: no cover
databricks_cli/dbfs/cli.py:def dbfs_group():  # pragma: no cover
databricks_cli/groups/cli.py:def groups_group():  # pragma: no cover
databricks_cli/instance_pools/cli.py:def instance_pools_group():  # pragma: no cover
databricks_cli/jobs/cli.py:def jobs_group():  # pragma: no cover
databricks_cli/libraries/cli.py:def libraries_group():  # pragma: no cover
databricks_cli/pipelines/cli.py:def pipelines_group():  # pragma: no cover
databricks_cli/runs/cli.py:def runs_group():  # pragma: no cover
databricks_cli/secrets/cli.py:def secrets_group():  # pragma: no cover
databricks_cli/stack/cli.py:def stack_group():  # pragma: no cover
databricks_cli/tokens/cli.py:def tokens_group():  # pragma: no cover
databricks_cli/workspace/cli.py:def workspace_group():  # pragma: no cover
```

nfx commented on databricks_cli/scim/exceptions.py:
```diff
@@ -0,0 +1,26 @@
+# limitations under the License.
+
+
+class ScimError(Exception):
```
move exception to respective file. why do we need a dedicated file for 2 lines of code?

nfx commented on databricks_cli/sdk/api_client.py:
```diff
@@ -127,6 +128,11 @@ def perform_query(self, method, path, data = {}, headers = None):
             raise requests.exceptions.HTTPError(message, response=e.response)
+
+        # delete returns nothing.
+        if method.lower() == 'delete':
```
not everywhere. please remove.

areese replied:
I'll have to play with this.
I think delete returns an empty body on the SCIM api, and that's the issue.

areese committed:
Merge branch 'master' into add-scim-api

areese commented on databricks_cli/click_types.py:
```diff
@@ -139,6 +142,30 @@ def __init__(self):
+        if debug:
+            self.enable_debug_logging()
+
+    def enable_debug_logging(self):
```
Fixed, I think this was an old left over bit, so it's removed.

areese replied:
removed instead, I don't think this is needed.

areese commented on databricks_cli/scim/api.py:
```diff
@@ -0,0 +1,269 @@
+
+    def list_users(self, filters=None, active=None):
+        # https://help.databricks.com/hc/en-us/requests/24688
+        # filtering on active doesn't work.
```
removed, but comment is kept.

areese committed:
Remove debug logging from the scim api that was only required during development
Fix delete method to only return an empty dict if there was no body in the response.

nfx reviewed (changes requested):
See change requests in context of specific methods.

nfx reviewed (changes requested):
See change requests in context of specific methods.

Please also add 85% unit test coverage for new code to match quality assurance criteria for other code in this repo - https://codecov.io/gh/databricks/databricks-cli/tree/master/databricks_cli

areese commented:
FYI: This is stale, it has semi-useful code but has a lot to be addressed.
I'm going to leave it open, but I'm not sure how fast I'll come back to it.

chrisst commented:
This repository has been archived and is no longer maintained. The legacy `databricks/databricks-cli` has been fully replaced by the new Databricks CLI at https://github.com/databricks/cli.
